_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-mailchimp.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * Mailchimp: Get Mailchimp Status.
+ * API to determine if current site has linked Mailchimp account and mailing list selected.
+ * This API is meant to be used in Jetpack and on WPCOM.
+ *
+ * @since 7.0
+ */
+class WPCOM_REST_API_V2_Endpoint_Mailchimp extends WP_REST_Controller {
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'mailchimp';
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+	/**
+	 * Called automatically on `rest_api_init()`.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
+			array(
+				array(
+					'methods'  => WP_REST_Server::READABLE,
+					'callback' => array( $this, 'get_mailchimp_status' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Get the status of current blog's Mailchimp connection
+	 *
+	 * @return mixed
+	 * code:string (connected|unconnected),
+	 * connect_url:string
+	 * site_id:int
+	 */
+	public function get_mailchimp_status() {
+		$is_wpcom    = ( defined( 'IS_WPCOM' ) && IS_WPCOM );
+		$site_id     = $is_wpcom ? get_current_blog_id() : Jetpack_Options::get_option( 'id' );
+		$connect_url = sprintf( 'https://wordpress.com/sharing/%s', $site_id );
+		$path        = sprintf( '/sites/%d/mailchimp/settings', $site_id );
+		if ( ! $is_wpcom ) {
+			$response = Jetpack_Client::wpcom_json_api_request_as_blog(
+				$path,
+				'1.1'
+			);
+			if ( is_wp_error( $response ) ) {
+				return new WP_Error( 'wpcom_connection_error', 'Could not connect to WP.com', 404 );
+			}
+			$data = isset( $response['body'] ) ? json_decode( $response['body'], true ) : null;
+		} else {
+			require_lib( 'mailchimp' );
+			$data = MailchimpApi::get_settings( $site_id );
+		}
+		$follower_list_id = isset( $data['follower_list_id'] ) ? $data['follower_list_id'] : null;
```

Comment:
So, as I understand it - we are NOT trying to do list selection inside the block right now?

Reply:
I am asking, since as I said - I have mixed feelings about this whole approach.
If this is a stopgap solution that we intend to change in the future, I would leave it at "set the option remotely" which we introduced in previous release.
If this is meant to stay, we have to think about longterm shape of this approach, particularly what endpoint is supposed to return.
This boils down to the fact, that we generally prefer keeping Jetpack site as a source of truth in options and I broke that pattern. Remote-setting kind of alleviates that since it at least ensures options are in sync.
But this is parting from the approach.
The endpoint you are using here to reach out to public-api endpoint is ingenious and I will definitely steal that code for memberships.
But I am on the fence what should happen in this particular case and getting someone from JP would be neat to resolve this.